src/components/GlobalPreferences/CustomLabels/Search.js

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Button,
+  ButtonIcon,
+  GU,
+  IconCross,
+  IconSearch,
+  TextInput,
+  useTheme,
+  useLayout,
+  textStyle,
+} from '@aragon/ui'
+
+const SEARCH_COMPACT = Symbol('compact')
+const SEARCH_OPEN = Symbol('open')
+const EMPTY = ''
+
+const Search = React.memo(function Search({ onChange, value }) {
+  const { layoutName } = useLayout()
+  const compact = layoutName === 'small'
+  const theme = useTheme()
+
+  const [mode, setMode] = useState(compact ? SEARCH_COMPACT : SEARCH_OPEN)
+  const open = useCallback(() => setMode(SEARCH_OPEN), [setMode])
+  const clear = useCallback(() => {
+    setMode(compact ? SEARCH_COMPACT : SEARCH_OPEN)
+    onChange({ currentTarget: { value: EMPTY } })
```

Comment:
Maybe we could just send the value, rather than mimicking the DOM event API? Since it would only work to access `currentTarget.value` exactly anyway.

Reply:
I thought on just sending an object `{ currentTarget: {value: EMPTY } }` but that's only helpful for the way we handle input events, what happens to other people who might be looking for other info from the event? This way, the actual event that bubbles is an input change.

Reply:
This was a simulation based on the current implementation, I plan on update to `SearchInput`'s api and functionality, having said that, let me know if you strongly feel I should make this change now and I can definitely do so.

Reply:
Never mind, this is actually different to `SearchInput` as it always needs to display the close button, so I'll take the suggestion and send the value without mimicking the dom event.